pydrawise/client.py

```diff
@@ -335,13 +335,35 @@ async def get_watering_report(
             self._schema.Controller.reports.select(
                 self._schema.Reports.watering(
                     **{
-                        "from": DateTime.to_json(start).timestamp,
-                        "until": DateTime.to_json(end).timestamp,
+                        "from": int(start.timestamp()),
+                        "until": int(end.timestamp()),
                     }
                 ).select(*get_selectors(self._schema, WateringReportEntry)),
             ),
         )
         result = await self._query(selector)
-        return deserialize(
+        entries = deserialize(
             list[WateringReportEntry], result["controller"]["reports"]["watering"]
         )
+        # The call to watering() can return events outside of the provided time interval.
+        # Filter out events that happen before or after the provided time interval.
+        return list(
+            filter(
+                lambda entry: entry.run_event is not None
+                and entry.run_event.reported_start_time is not None
+                and entry.run_event.reported_end_time is not None
+                and (
+                    (
+                        start.timestamp()
```

Comment:
These should all be native datetimes so you should be able to compare them directly without calling `timestamp()` on each one.

Reply:
That's how I started originally. The problem is that the datetime that is coming from HA in the arguments is timezone-aware, but the datetime that we parse from the Hydrawise result is not. This results in a "can't compare offset-naive and offset-aware datetimes" error. We would have to change `default_datetime()` in `schema.py` first to make it timezone aware.